portfolio/src/main/webapp/script.js

```diff
@@ -99,7 +99,7 @@ function questions() {
  */
 function updateComments() {
     var numComments = document.getElementById("num-comments").value;
-    if(numComments < 1 || !Number.isInteger(numComments)) {
+    if(numComments < 1 || (numComments - Number.parseInt(numComments)) !== 0) {
```

Comment:
In what way is Number.isInteger unreliable? Can you actually identify what input results it it returning a value other than what you expect?

Consider that there may be something else your code is doing that doesn't match what you expect. For example, what is the type of numComments? Are you sure it's actually a number? If you call `isInteger` on a string, what happens?

Reply:
I use it to trigger an alert when the user enters a decimal value or a value < 1. The page automatically loads a default of 10 and once the page loads, it triggers the alert. When I try to display 8,5,6,7, etc it triggers the same alert. I even removed the not operator, but that option accepts both ints and floats.

I made sure the textbox had an input type of number and I can't type in letters (except the char e, not sure why). Is there another way to determine if it is a number and not string?

Reply:
When you set the input type in HTML, that affects how the control is rendered and applies validation to it so you can't input an invalid value, but it doesn't actually affect the type of the value stored in the input field.

If you want to know the type that something has in JavaScript, you can use `typeof value` which returns a string that tells you what type the value is, for example, you could use `console.log(typeof numComments);` or `alert(typeof numComments);`

Reply:
ahh ok I see now, I assumed it was of type number since the other approach I took worked. I just confirmed it with my local server and I'll fix it, thanks